### PR TITLE
Move tests to mock server

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Move GetDataFromPartitionSync and
GetDataFromPartitionAsync to work
on mock server. Add helper class.
Move host and port to environment
variables.

Relates-To: OLPEDGE-1971

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+// clang-format on
+#include "ApiDefaultResponses.h"
+#include "MockServerHelper.h"
+#include "ReadDefaultResponses.h"
+#include "Utils.h"
+
+namespace {
+
+const auto kMockServerHost = "localhost";
+const auto kMockServerPort = 1080;
```
Comment (meravingen123) [inline tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp:36]: Maybe we can pass these as environment variables?

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    auto network = std::move(settings_->network_request_handler);
+    settings_.reset();
+    mock_server_client_.reset();
+  }
+
+  std::shared_ptr<olp::client::OlpClientSettings> settings_;
+  std::shared_ptr<mockserver::MockServerHelper> mock_server_client_;
+};
+
+TEST_F(VersionedLayerClientGetPartitionTest, GetDataFromPartitionSync) {
```
Comment (kzvieriev) [inline tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp:64]: Why test GetData() in file for GetPartition() api

File: scripts/linux/fv/olp-cpp-sdk-functional-test.variables [hunk 3]
```diff
@@ -31,3 +31,6 @@ export dataservice_read_volatile_test_prefetch_appid="${dataservice_read_volatil
 export dataservice_read_volatile_test_prefetch_secret="${dataservice_read_volatile_test_prefetch_secret}"
 export dataservice_read_volatile_test_prefetch_catalog="hrn:here:data::olp-here-test:test-volatile-prefetch"
 export dataservice_read_volatile_prefetch_layer="volatile_prefetch_layer"
+
+export mock_server_host="localhost"
+export mock_server_port="1080"
```
Comment (kzvieriev) [inline scripts/linux/fv/olp-cpp-sdk-functional-test.variables:36]: add new line to the file end

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      olp::dataservice::read::DataRequest().WithPartitionId(partition));
+  auto response = future.GetFuture().get();
+  EXPECT_SUCCESS(response);
+  ASSERT_TRUE(response.GetResult() != nullptr);
+  ASSERT_EQ(response.GetResult()->size(), data.size());
+
+  EXPECT_TRUE(mock_server_client_->Verify());
+}
+
+TEST_F(VersionedLayerClientGetPartitionTest, GetDataFromPartitionAsync) {
```
Comment (kzvieriev) [inline tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp:96]: same here, move to GetData() file

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+class VersionedLayerClientGetPartitionTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    auto network = olp::client::OlpClientSettingsFactory::
+        CreateDefaultNetworkRequestHandler();
+    settings_ = mockserver::SetupMockServer::CreateSettings(network);
+    mock_server_client_ =
+        mockserver::SetupMockServer::CreateMockServer(network, kTestHrn);
+  }
```
Comment (kzvieriev) [inline tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp:52]: Its better now, but still need to copy whole SetUp() for each api test and each client.
I think we need base class for all mock server tests (like E2ETestBase for e2e tests).

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+const auto kTestHrn = "hrn:here:data::olp-here-test:hereos-internal-test";
+const auto kLayer = "testlayer";
+const auto kVersion = 44;
+constexpr auto kWaitTimeout = std::chrono::seconds(10);
+const auto kPartitionsResponsePath =
+    "/query/v1/catalogs/hrn:here:data::olp-here-test:hereos-internal-test/"
+    "layers/testlayer/partitions";
+
+class VersionedLayerClientGetPartitionTest : public ::testing::Test {
```
Comment (kzvieriev) [inline tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp:44]: Its cool to make 1 file per api, but name becomes too complicated.
Maybe we can create 1 test class/alias for all versioned client tests, so test names are readable, e.g. TEST_F(VersionedLayerClient, GetPartitionsAsync)

File: scripts/linux/fv/gitlab-olp-cpp-sdk-functional-test-mock.sh [hunk 7]
```diff
@@ -51,7 +51,7 @@ update-ca-certificates
 echo ">>> Starting Functional Test against Mock Server... >>>"
 $REPO_HOME/build/tests/functional/olp-cpp-sdk-functional-tests \
     --gtest_output="xml:$REPO_HOME/reports/olp-functional-test-mock-report.xml" \
-    --gtest_filter="VersionedLayerClientTest.GetPartitions":"VersionedLayerClientTest.GetAggregatedData":"CatalogClientTest.*":"VersionedLayerClientPrefetchTest.Prefetch":"VersionedLayerClientProtectTest.*"
+    --gtest_filter="VersionedLayerClientTest.GetPartitions":"VersionedLayerClientTest.GetAggregatedData":"CatalogClientTest.*":"VersionedLayerClientPrefetchTest.Prefetch":"VersionedLayerClientProtectTest.*":"VersionedLayerClientGetPartitionTest.*"
```
Comment (kzvieriev) [inline scripts/linux/fv/gitlab-olp-cpp-sdk-functional-test-mock.sh:54]: Test name changed :-)

**Review by kzvieriev — APPROVED** [review]

**Review by meravingen123 — APPROVED** [review]

**Pull request merged**